src/PowerShellEditorServices/Session/PowerShellContext.cs

```diff
@@ -1113,30 +1113,143 @@ public async Task SetWorkingDirectory(string path, bool isPathAlreadyEscaped)
+        }
+
+        /// <summary>
+        /// Wrap an already escaped path in quotes to make it safe to use in scripts.
+        /// </summary>
+        /// <param name="escapedPath">The glob-escaped path to wrap in quotes.</param>
+        /// <returns>The given path wrapped in quotes appropriately.</returns>
+        internal static string QuoteEscapePath(string escapedPath)
+        {
+            var sb = new StringBuilder(escapedPath.Length + 2); // Length of string plus two quotes
+            sb.Append('\'');
+            if (!escapedPath.Contains('\''))
+            {
+                sb.Append(escapedPath);
+            }
+            else
+            {
+                foreach (char c in escapedPath)
+                {
+                    if (c == '\'')
+                    {
+                        sb.Append("''");
+                        continue;
+                    }
+
+                    sb.Append(c);
+                }
+            }
+            sb.Append('\'');
+            return sb.ToString();
+        }
+
+        /// <summary>
+        /// Return the given path with all PowerShell globbing characters escaped,
+        /// plus optionally the whitespace.
+        /// </summary>
+        /// <param name="path">The path to process.</param>
+        /// <param name="escapeSpaces">Specify True to escape spaces in the path, otherwise False.</param>
+        /// <returns>The path with [ and ] escaped.</returns>
+        internal static string GlobEscapePath(string path, bool escapeSpaces = false)
+        {
+            var sb = new StringBuilder();
+            for (int i = 0; i < path.Length; i++)
+            {
+                char curr = path[i];
+                switch (curr)
+                {
+                    // Escape '[', ']', '?' and '*' with '`'
+                    case '[':
+                    case ']':
+                    case '*':
+                    case '?':
+                        sb.Append('`').Append(curr);
+                        break;
+
+                    default:
+                        // Escape whitespace if required
+                        if (escapeSpaces && char.IsWhiteSpace(curr))
+                        {
+                            sb.Append('`').Append(curr);
+                            break;
+                        }
+                        sb.Append(curr);
+                        break;
+                }
+            }
+
+            return sb.ToString();
+        }
+
         /// <summary>
         /// Returns the passed in path with the [ and ] characters escaped. Escaping spaces is optional.
         /// </summary>
         /// <param name="path">The path to process.</param>
         /// <param name="escapeSpaces">Specify True to escape spaces in the path, otherwise False.</param>
         /// <returns>The path with [ and ] escaped.</returns>
+        [Obsolete("This API is not meant for public usage and should not be used.")]
```

Comment:
I'd add this above `Obsolete`

```csharp
[Hidden, EditorBrowsable(EditorBrowsableState.Never)]
```

`EditorBrowsable` hides it from intellisense in external C# libraries, `Hidden` hides it from intellisense in PowerShell.

Reply:
Looks like `HiddenAttribute` isn't in PowerShell v3/4?

Reply:
Bah! You're right :/ it was added for classes